commit 1c08f511c27beb9cf9d0fd4902034e408a098cab (refs/changes/18/1418/1)
Author: andrew <andrew@whartonlabs.com>
Date:   2017-02-03T13:25:19+11:00 (2 years, 8 months ago)
    
    Added integration test for resizing instance.
    
    Also had to add an additional config for the package to resize to as well as a helper to grab this info.

diff --git a/README.md b/README.md
index f8d67e6..0fe770a 100644
--- a/README.md
+++ b/README.md
@@ -368,7 +368,8 @@ test-integration`). Integration tests require a config file, by default at
         "profileName": "east3b",
         "allowWriteActions": true,
         "image": "minimal-64",
-        "package": "t4-standard-128M"
+        "package": "g4-highcpu-128M",
+        "resizePackage": "g4-highcpu-256M"
     }
 
 See "test/config.json.sample" for a description of all config vars. Minimally
diff --git a/test/config.json.sample b/test/config.json.sample
index 44e2e65..3f59cc5 100644
--- a/test/config.json.sample
+++ b/test/config.json.sample
@@ -25,5 +25,6 @@
     // The params used for test provisions. By default the tests use:
     // the smallest RAM package, the latest base* image.
     "package": "<package name or uuid>",
+    "resizePackage": "<package name or uuid>",
     "image": "<image uuid, name or name@version>"
 }
diff --git a/test/integration/cli-manage-workflow.test.js b/test/integration/cli-manage-workflow.test.js
index 0d5eed1..edf3037 100644
--- a/test/integration/cli-manage-workflow.test.js
+++ b/test/integration/cli-manage-workflow.test.js
@@ -68,6 +68,15 @@ test('triton manage workflow', opts, function (tt) {
         });
     });
 
+    var resizePkgName;
+    tt.test('  setup: find resize test package', function (t) {
+        h.getResizeTestPkg(t, function (err, pkgName_) {
+            t.ifError(err, 'getResizeTestPkg' + (err ? ': ' + err : ''));
+            resizePkgName = pkgName_;
+            t.end();
+        });
+    });
+
     // create a test machine (blocking) and output JSON
     tt.test('  setup: triton create', function (t) {
         var argv = [
@@ -237,6 +246,27 @@ test('triton manage workflow', opts, function (tt) {
         });
     });
 
+    // resize the instance
+    tt.test('  triton inst resize', function (t) {
+        var args = ['inst', 'resize', '-w', instance.id, resizePkgName];
+        h.safeTriton(t, args, function (err, stdout) {
+            t.ok(stdout.match(/^Resizing instance/m),
+              '"Resizing instance" in stdout');
+            t.ok(stdout.match(/^Resized instance/m),
+              '"Resized instance" in stdout');
+            t.end();
+        });
+    });
+
+    tt.test('  confirm resized', function (t) {
+        h.safeTriton(t, {json: true, args: ['inst', 'get', '-j',
+              INST_ALIAS]},
+          function (err, inst) {
+              t.equal(inst.package, resizePkgName, 'instance was resized');
+              t.end();
+          });
+    });
+
     // rename the instance
     tt.test('  triton inst rename', function (t) {
         var args = ['inst', 'rename', '-w', instance.id, INST_ALIAS_NEWNAME];
diff --git a/test/integration/helpers.js b/test/integration/helpers.js
index 0e096fc..cb4a858 100644
--- a/test/integration/helpers.js
+++ b/test/integration/helpers.js
@@ -231,6 +231,31 @@ function getTestPkg(t, cb) {
     });
 }
 
+/*
+ * Find and return second smallest package name that can be used for test provisions.
+ *
+ * @param {Tape} t - tape test object
+ * @param {Function} cb - `function (err, {pkgs})`
+ *      where `pkgs` is an Array of 2 test packages to use.
+ */
+function getResizeTestPkg(t, cb) {
+    if (CONFIG.resizePackage) {
+        t.ok(CONFIG.resizePackage, 'resizePackage from config: ' + CONFIG.resizePackage);
+        cb(null, CONFIG.resizePackage);
+        return;
+    }
+
+    safeTriton(t, ['pkg', 'ls', '-j'], function (err, stdout) {
+        var pkgs = jsonStreamParse(stdout);
+        // Smallest RAM first.
+        tabula.sortArrayOfObjects(pkgs, ['memory']);
+        var pkg = pkgs[1];
+        t.ok(pkg.name, f('second smallest (RAM) available package: %s (%s)',
+          pkg.id, pkg.name));
+        cb(null, pkg.name);
+    });
+}
+
 
 function jsonStreamParse(s) {
     var results = [];
@@ -344,6 +369,7 @@ module.exports = {
     deleteTestInst: deleteTestInst,
     getTestImg: getTestImg,
     getTestPkg: getTestPkg,
+    getResizeTestPkg: getResizeTestPkg,
     jsonStreamParse: jsonStreamParse,
     printConfig: printConfig,
 
